### PR TITLE
COL-906 Periodically recalculate asset trending scores from recent activity

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/COL-906

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -49,6 +49,10 @@
     "apiKey": "some shared secret"
   },
   "timezone": "America/Los_Angeles",
+  "trendingScores": {
+    "activityWindow": 168,
+    "recalculationInterval": 60
```
Comment (paulkerschen) [inline config/default.json]: As noted in comments below, `activityWindow` uses hour units and `recalculationInterval` uses minutes. The idea was to make it easy to override in QA.

File: node_modules/col-activities/lib/api.js [hunk 2]
```diff
@@ -1006,6 +1023,92 @@ var setImpactScore = function(score, assetId, callback) {
...
+  if (course) {
+    log.info({'course': course.id}, 'Will update trending scores for a single course');
+    activityOptions.where.course_id = course.id;
+  } else {
+    log.info('Will update trending scores for all courses');      
+  }
+  
+  DB.Activity.findAll(activityOptions).complete(function(err, activities) {
+    if (err) {
+      log.error({'err': err}, 'Failed to get activities for the course');
```
Comment (johncrossman) [inline node_modules/col-activities/lib/api.js]: @paulkerschen minor detail: Should error message above be sensitive to 'single course' vs. 'all courses'?

Comment (paulkerschen) [inline node_modules/col-activities/lib/api.js]: @johncrossman Yes, thanks; amended to log query options as they may come.

**Comment by johncrossman** [comment]
Nice work, @paulkerschen  Let the trending begin

**Pull request merged**